components/raftstore/src/store/fsm/peer.rs

```diff
@@ -4417,13 +4417,13 @@ where
 
         let first_idx = self.fsm.peer.get_store().first_index();
 
-        let mut compact_idx = if force_compact
-            // Too many logs between applied index and first index.
-            || (applied_idx > first_idx && applied_idx - first_idx >= self.ctx.cfg.raft_log_gc_count_limit)
-            // Raft log size ecceeds the limit.
+        let mut compact_idx = if force_compact && replicated_idx > first_idx {
```

Comment:
For the lag log issue, why RocksDB works without this change?

Reply:
Only raft engine triggers force compaction, so there won’t be any unexpected  side effect.

Reply:
> Only raft engine triggers force compaction, so there won’t be any unexpected side effect.
IC. It will be better to add comment on parameter 'force_compact' that only raft engine uses this flag.